### ISSUE TITLE
DagPut broken in 1.2.0

### ISSUE THREAD

**Issue by bclermont** [ipfs/go-ipfs-api#64]
I just upgraded this repo to 1.2.0 to fix the `AddDir` bug and now:

```
package main

import (
	"log"

	"github.com/ipfs/go-ipfs-api"
)

func main() {
	s := shell.NewLocalShell()
	obj, err := s.DagPut(`{"x": "abc","y":"def"}`, "json", "cbor")
	if err != nil {
		log.Panic(err)
	}
	log.Println(obj)
}
```

```
2017/11/23 22:41:31 dag/put: File argument 'object data' is required
panic: dag/put: File argument 'object data' is required
```

I haven't replaced the daemon. Still running the same version

**Comment by bclermont** [comment]
I just tried with go-ipfs 0.4.2, same error

**Comment by magik6k** [comment]
What was the IPFS version for the first try?

**Comment by bclermont** [comment]
> What was the IPFS version for the first try?

0.4.11

**Comment by bclermont** [comment]
Tried on a empty `GOPATH` and `go get github.com/ipfs/go-ipfs-api`, same

**Comment by lgierth** [comment]
@bclermont fixed, and released 1.2.1 -- I overlooked a tiny thing in #60 that I fixed in #65

**Comment by bclermont** [comment]
I confirm it's fixed!
